File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.cpp
class Solution {
public:
    int largestRectangleArea(vector<int>& heights) {
        vector<pair<int, int>> pair;
        int maxArea = 0;

        for(int i=0; i<heights.size(); i++){
            int start = i;
            while(!pair.empty() && heights[i] < pair.back().first){
                start = pair.back().second;
                int mul = i - pair.back().second;
                int area = pair.back().first * mul;
                maxArea = max(maxArea, area);
                pair.pop_back();
            }
            pair.push_back({heights[i], start});
        }

        int maxI = heights.size();
        for(auto p: pair){
            int area = p.first * (maxI - p.second);
            maxArea = max(maxArea, area);
        }
        return maxArea;
    }
};

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-1.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        mx, my, mz = 0, 0, 0
        for x, y, z in triplets:
            if x <= target[0] and y <= target[1] and z <= target[2]:
                mx = max(x, mx)
                my = max(y, my)
                mz = max(z, mz)
        
        return [mx, my, mz] == target